exporter/datadogexporter/translate_traces_test.go

```diff
@@ -1456,3 +1456,59 @@ func TestSpanEnvClobbering(t *testing.T) {
 	assert.Equal(t, "correctenv", outputTraces[0].Traces[0].Spans[0].Meta["env"])
 	assert.Equal(t, "correctenv", outputTraces[0].Traces[0].Spans[0].Meta["deployment.environment"])
 }
+
+// ensure that global rate limit tag gets set as a metric
+func TestSpanRateLimitTag(t *testing.T) {
+	mockTraceID := [16]byte{0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F}
+	mockSpanID := [8]byte{0xF1, 0xF2, 0xF3, 0xF4, 0xF5, 0xF6, 0xF7, 0xF8}
+	mockParentSpanID := [8]byte{0xEF, 0xEE, 0xED, 0xEC, 0xEB, 0xEA, 0xE9, 0xE8}
+	endTime := time.Now().Round(time.Second)
+	pdataEndTime := pdata.TimestampFromTime(endTime)
+	startTime := endTime.Add(-90 * time.Second)
+	pdataStartTime := pdata.TimestampFromTime(startTime)
+
+	denylister := newDenylister([]string{})
+	buildInfo := component.BuildInfo{
+		Version: "1.0",
+	}
+
+	traces := pdata.NewTraces()
+	traces.ResourceSpans().EnsureCapacity(1)
+	rs := traces.ResourceSpans().AppendEmpty()
+	resource := rs.Resource()
+	resource.Attributes().InitFromMap(map[string]pdata.AttributeValue{
+		conventions.AttributeDeploymentEnvironment: pdata.NewAttributeValueString("correctenv"),
+		"env": pdata.NewAttributeValueString("incorrectenv"),
+	})
+
+	rs.InstrumentationLibrarySpans().EnsureCapacity(1)
+	ilss := rs.InstrumentationLibrarySpans().AppendEmpty()
+	instrumentationLibrary := ilss.InstrumentationLibrary()
+	instrumentationLibrary.SetName("flash")
+	instrumentationLibrary.SetVersion("v1")
+	span := ilss.Spans().AppendEmpty()
+
+	attribs := map[string]pdata.AttributeValue{
+		"_sample_rate": pdata.NewAttributeValueString("0.5"),
+	}
+
+	span.Attributes().InitFromMap(attribs)
+
+	traceID := pdata.NewTraceID(mockTraceID)
+	spanID := pdata.NewSpanID(mockSpanID)
+	parentSpanID := pdata.NewSpanID(mockParentSpanID)
+	span.SetTraceID(traceID)
+	span.SetSpanID(spanID)
+	span.SetParentSpanID(parentSpanID)
+	span.SetName("End-To-End Here")
+	span.SetKind(pdata.SpanKindServer)
+	span.SetStartTimestamp(pdataStartTime)
+	span.SetEndTimestamp(pdataEndTime)
+
+	config := config.Config{Traces: config.TracesConfig{SpanNameRemappings: map[string]string{"flash.server": "bang.client"}}}
+
+	outputTraces, _ := convertToDatadogTd(traces, "test-host", &config, denylister, buildInfo)
+
+	// Ensure the deployment.environment value is copied to both deployment.environment and env
```

Comment:
I think this comment doesn't apply here.

Reply:
👍 updated